deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -21,10 +21,11 @@ class ChatViewController: UITableViewController {
     // The VC can be dismissed by pressing back '<' or by a swipe-to-dismiss gesture.
     // The latter is cancelable and leads to viewWillAppear is called in case the gesture is cancelled
     // We need the flag to handle that special case correctly in viewWillAppear
-    var isDismissing = false
-    var isInitial = true
-    var ignoreInputBarChange = false
+    private var isDismissing = false
+    private var isInitial = true
+    private var ignoreInputBarChange = false
     private var isVisibleToUser: Bool = false
+    private var keepKeyboard: Bool = false
```

Comment:
maybe add a little comment what keepKeyboard actually does.

iiuc, it is set on sending and sets the focus to the input bar on sending so that it is not hidden?

Reply:
right, it indicates that the currently focussed input field should not loose and re-request it's focus. The visibility of the keyboard is always related to the focus of an editable input field in iOS. There's no other way to control the keyboard than by requesting to become the first responder or by resigning that. The evaluation if a text input field should loose it's focus can be done in the delegate method `textViewShouldEndEditing`